test/unit/specs/validators/email.spec.js

```diff
@@ -0,0 +1,48 @@
+import email from 'src/validators/email'
+
+describe('email validator', () => {
+  it('should not validate empty string', () => {
+    expect(email('')).to.be.false
+  })
+
+  it('should not validate numbers', () => {
+    expect(email('12345')).to.be.false
+  })
+
+  it('should not validate strings', () => {
+    expect(email('asdf12345')).to.be.false
+  })
+
+  it('should not validate space', () => {
+    expect(email(' ')).to.be.false
+  })
+
+  it('should not validate incomplete addresses', () => {
+    expect(email('someone@')).to.be.false
+    expect(email('someone@gmail')).to.be.false
+    expect(email('someone@gmail.')).to.be.false
+    expect(email('someone@gmail.c')).to.be.false
+    expect(email('gmail.com')).to.be.false
+    expect(email('@gmail.com')).to.be.false
+  })
+
+  it('should not validate addresses that contain unsupported characters', () => {
+    expect(email('someone@g~mail.com')).to.be.false
+    expect(email('some!one@gmail.com')).to.be.false
```

Comment:
At work we tend to settle on something like the following "casual" approach…

`[characters, no spaces]@[characters, no spaces].[characters, no spaces]`

…which helps prevent typos, which generally suits our immediate front-end needs.

For complete validation, as @Frizi suggested, after "casual" validation we'd ping a backend validation API (and occasionally follow up with a confirmation email, depending on business requirements).

Cheers. 🙂